### PR TITLE
Add completions for mdadm

### PR DESCRIPTION
**Pull request by PenegalECI**
## Description

This PR adds completions for `mdadm`, the Linux tool for maintaining software RAID. This has been developed using the `man` page, so it should be rather exhaustive, taking into account around 98% of use cases AFAIK, but there could be some lacks due to:
* `mdadm` idiosyncrasies, like using the same short option for several long ones, or their different meanings in different contexts,
* a misunderstanding of the `man`, being only a regular `mdadm` user, not an expert,
* some specific combination of tags,
* remaining bugs or typos, although I checked my code along the whole coding.

## TODOs:

- [ ] Changes to fish usage are reflected in user documentation/manpages.
- [ ] Tests have been added for regressions fixed
- [ ] User-visible changes noted in CHANGELOG.md

### CODE REVIEW

File: share/completions/mdadm.fish [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+# - especially, suggest values for levels, metadata format… according to the already given parameters
+# - some descriptions are not very useful, even a bit Captain Obvious-like
+# - some options could see their meaning in different contexts explained with different completions with proper tests for context
+# - misc and manage modes could be tested for, and the corresponding options proposed only in the right context
+#
+# -n tests have been simplified when __fish_contains_opt …; or _fish_not_contain_opt …: the options in the first test were removed from the second one,
+#       as their absence has already been checked, because this absence caused evaluation of __fish_not_contain_opt
+
+function __fish_mdadm_metadata_options
+	echo "0	Use original 0.90 format superblock"
```
Comment (faho) [inline share/completions/mdadm.fish]: Are these literal tab characters?

I'd prefer it if we used the `\t` escape instead.

It's even possible to use something like

```fish
printf '%s\t%s\n' "0" "Use original 0.90 format superblock" \
                          "0.90" "Use original 0.90 format superblock"
# and so on
```

Comment (faho) [inline share/completions/mdadm.fish]: Or, a bit simpler:

```fish
echo "0"\t"Use original 0.90 format superblock"
echo "0.90"\t"..."
```

or countless variations. Anyway, please use `\t`, no tab characters in our files if we can help it.

**Comment by krader1961** [comment]
I agree with @faho. Do not include literal tabs (or other non-printable chars) in those strings since they are effectively invisible. Please use a symbolic representation like `\t` as @faho suggested so that it is obvious to anyone reading the script they are present.

**Comment by PenegalECI** [comment]
Here you are.

**Comment by faho** [comment]
And there you are - merged, thanks!

**Pull request merged**